### PR TITLE
Update CI workflow and add code analysis

### PR DESCRIPTION
**Pull request by sam-baraka**
This pull request updates the CI workflow for the Flutter Con Kenya 2024 mobile application. It adds a step to analyze the code using the `flutter analyze` command. This step ensures that the code follows best practices and identifies any potential issues or errors. Additionally, the pull request includes the necessary configuration to build the APK and upload it as an artifact. Finally, it adds a step to create a GitHub release if the push event is a tag push. The release includes the uploaded APK as an artifact and includes a description mentioning the tag name. This pull request improves the development process by automating the build and release tasks and ensuring code quality.

### CODE REVIEW

File: .github/main.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: android-app
+          path: build/app/outputs/apk/release/app-release.apk  # Update path if needed
+
+      - name: Create GitHub Release (if tag)
+        uses: ncipollo/release-action@v1
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')  # Only on tag push
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}  # Access token with release permissions
```
Comment (sam-baraka) [inline .github/main.yml:42]: @manuelgeek we can have an Access token with release permissions created in the repository

Comment (manuelgeek) [reply .github/main.yml:42]: I think default `github.token` should be enough?

Comment (sam-baraka) [reply .github/main.yml:42]: You're right

**Comment by sam-baraka** [comment]
> I think default `github.token` should be enough? 

You're right, thanks

**Review by NewtonMutugi — APPROVED** [review]

**Review by KendiJ — APPROVED** [review]

**Pull request merged**